src/content/en/fundamentals/engage-and-retain/push-notifications/sending-messages.markdown

```diff
@@ -3,93 +3,343 @@ layout: shared/narrow
+  
+The first part (`dh=applicationServerKey`) is the public key, the one we created 
+under Requesting permission and subscribing users. The second part 
+(`p256ecdsa=publicKey`) is the public key, but base64 url encoded. Notice the 
+comma separating the two parts of the crypto-key.  
+  
+**Note:** A bug in Chrome 52 requires that a semicolon be send to the message 
+server in the Crypto-Key header instead of a comma.
+
+### Authorization header
+
+The standards-based way of sending a message is through a system called 
+Voluntary Application Server Identification for Web Push or VAPID. This article 
+is written with the assumption that VAPID is how your app server will send 
+messages to your client. Everything we've covered so far applies to message 
+servers that don't use VAPID (though, those methods may have additional 
+requirements of their own). 
+
+To use VAPID, you need an an Authorization header. It contains four parts:  
+  
+<pre>Bearer &lt;JWTHeader&gt;.&lt;Payload&gt;.&lt;Signature&gt;</pre>
+
+The word Bearer is a literal that must be followed by a space. The remaining 
+portions, which are encrypted and concatenated using a period are a signed JSON 
+web token (JWT). A JWT is a way of sharing a JSON object with a second party in 
+such a way that the sending party can sign it and the receiving party can verify 
+the signature is from the expected sender. Adding the literal Bearer to the 
+front makes it a bearer token.  
+  
+You must encrypt all messages before sending them. Encryption is enough of a 
+specialty, even within software development, that we don't recommend writing 
+your own encryption system. Fortunately, there are a range of push libraries 
+including our own Push Encryption Library.  
+  
+Let's look at each part of the bearer token in detail.
+
+#### JWT Header
+
+The JWT Header contains two standard pieces of information: a typ property to 
+indicate that this is a JWT message, and an alg property to indicate the 
+algorithm used for encrypting the rest of the message. For VAPID, the later must 
+be base64 url encoded.
+
+{% highlight javascript %} 
+{  
+  "typ": "JWT",  
+  "alg": "ES256"  
+}
+{% endhighlight %}
+
+#### JWT Payload
+
+The JWT calls this section the payload. This isn't where the message payload is 
+stored. We'll get to that shortly. The payload is another JSON object with the 
+following members:    
+**aud**  
+This contains the push service endpoint, which you should extract from the 
+subscription object. This is not the origin of your site.    
+**exp**  
+Specifies the time the request expires in milliseconds. It must be within 
+twenty-four hours. This can be calculated by converting the current date to 
+milliseconds and adding the duration. For example, in Node.js you could do this: 
+   
+{% highlight javascript %}
+Math.floor((Date.now() / 1000) + 12 * 60 * 60)
+{% endhighlight %}
+
+This should match the value in the TTL header.
```

Comment:
Where did this come from?

The expiration here is the expiration of the JWT, which is seperate from the expiration of sending a message and how long it can live on a server.